modules/apps/captcha/captcha-rest-impl/src/main/java/com/liferay/captcha/rest/internal/resource/v1_0/CaptchaResourceImpl.java

```diff
@@ -0,0 +1,151 @@
+import com.liferay.portal.kernel.captcha.CaptchaTextException;
+import com.liferay.portal.kernel.encryptor.EncryptorUtil;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.module.configuration.ConfigurationProvider;
+import com.liferay.portal.kernel.util.Base64;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.io.ByteArrayOutputStream;
+
+import javax.servlet.http.HttpServletRequest;
+
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.ForbiddenException;
+import javax.ws.rs.NotAcceptableException;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+
+/**
+ * @author Loc Pham
+ */
+@Component(
+	properties = "OSGI-INF/liferay/rest/v1_0/captcha.properties",
+	scope = ServiceScope.PROTOTYPE, service = CaptchaResource.class
+)
+public class CaptchaResourceImpl extends BaseCaptchaResourceImpl {
+
+	@Override
+	public com.liferay.captcha.rest.dto.v1_0.Captcha getSimpleCaptcha()
+		throws Exception {
+
+		_checkSimpleCaptchaConfiguration();
+
+		Captcha captcha = CaptchaUtil.getCaptcha();
+
+		ByteArrayOutputStream imageByteArrayOutputStream =
+			new ByteArrayOutputStream();
+
+		String captchaAnswer = captcha.serveImageOutputStream(
+			_httpServletRequest, imageByteArrayOutputStream);
+
+		String base64CaptchaImage =
+			"data:image/png;base64," +
+				Base64.encode(imageByteArrayOutputStream.toByteArray());
+
+		imageByteArrayOutputStream.close();
+
+		return new com.liferay.captcha.rest.dto.v1_0.Captcha() {
+			{
+				captchaToken = CaptchaTokenUtil.generateCaptchaToken(
+					_contextCompany, captchaAnswer);
+
+				image = base64CaptchaImage;
+			}
+		};
+	}
+
+	@Override
+	public FormCaptcha validateSimpleCaptcha(FormCaptcha formCaptcha)
+		throws Exception {
+
+		_checkSimpleCaptchaConfiguration();
+
+		try {
+			CaptchaTokenUtil.checkAnswer(
+				_contextCompany, formCaptcha.getCaptchaToken(),
+				formCaptcha.getAnswer());
+		}
+		catch (CaptchaTextException captchaTextException) {
+			ByteArrayOutputStream imageByteArrayOutputStream =
+				new ByteArrayOutputStream();
+
+			Captcha simpleCaptcha = CaptchaUtil.getCaptcha();
+
+			String captchaAnswer = simpleCaptcha.serveImageOutputStream(
+				_httpServletRequest, imageByteArrayOutputStream);
+
+			String base64CaptchaImage =
+				"data:image/png;base64," +
+					Base64.encode(imageByteArrayOutputStream.toByteArray());
+
+			imageByteArrayOutputStream.close();
+
+			com.liferay.captcha.rest.dto.v1_0.Captcha captcha =
+				new com.liferay.captcha.rest.dto.v1_0.Captcha();
+
+			captcha.setCaptchaToken(
+				EncryptorUtil.encrypt(
+					_contextCompany.getKeyObj(),
+					CaptchaTokenUtil.generateCaptchaToken(
+						_contextCompany, captchaAnswer)));
+
+			captcha.setImage(base64CaptchaImage);
```

Comment:
Could refactor these codes to 
`com.liferay.captcha.rest.dto.v1_0.Captcha captcha = getSimpleCaptcha();`